homeassistant/components/pilight/light.py

```diff
@@ -61,7 +61,18 @@ def supported_features(self):
 
     def turn_on(self, **kwargs):
         """Turn the switch on by calling pilight.send service with on code."""
-        self._brightness = kwargs.get(ATTR_BRIGHTNESS, 255)
-        dimlevel = int(self._brightness / (255 / self._dimlevel_max))
+        # Update brightness only if provided as an argument, this will allow the switch to keep its previous brightness level.
```

Comment:
```suggestion
        # Update brightness only if provided as an argument.
        # This will allow the switch to keep its previous brightness level.
```